.github/workflows/report-failure.yaml

```diff
@@ -0,0 +1,87 @@
+# This workflow is intended to be called by workflows in our various software
+# repos to have shared workflow failure reporting in the form of GitHub issues.
+on:
+  workflow_call:
+    inputs:
+      failure:
+        description: Whether to create an issue indicating a failed workflow run, or keep an existing issue open.
+        type: boolean
+        required: true
+      success:
+        description: Whether to close any open issues created by this workflow.
+        type: boolean
+        required: true
+      title:
+        description: Title for issues opened by this workflow.
+        type: string
+        required: true
+    secrets:
+      token:
+        description: A GitHub access token with permission to create and modify GitHub issues.
+        required: true
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Extract workflow link
+        run: |
+          workflow_path=$(echo "${{ github.workflow_ref }}" | sed -E 's|^.*/(\.github/workflows/[^@]+)@\S*$|\1|')
+          echo "workflow_link=https://github.com/${{ github.repository }}/blob/${{ github.sha }}/$workflow_path" >> "$GITHUB_ENV"
+
+      - name: Create or close issue
+        run: |
+          # Search for issues opened by this job.
+          matching_issues=$(gh issue list \
+            --state open \
+            --author '@me' \
+            --json 'number,title' \
+            --jq 'map(select(.title == "${{ inputs.title }}") | .number)')
```

Comment:
`inputs.title` (here and below) should be mapped to an environment variable (e.g. `env: TITLE: ${{ inputs.title }}`) and accessed as such in the jq code (e.g. `env.TITLE`). Otherwise quotes in the title cause havoc.